packages/airnode-node/src/types.ts

```diff
@@ -206,33 +214,39 @@ export interface AuthorizationByRequestId {
   readonly [requestId: string]: boolean;
 }
 
-export type ApiCallResponse = ApiCallSuccessResponse | ApiCallErrorResponse;
+export type RegularApiCallResponse = RegularApiCallSuccessResponse | ApiCallErrorResponse;
+export type HttpGatewayApiCallResponse = HttpGatewayApiCallSuccessResponse | ApiCallErrorResponse;
+export type HttpSignedDataApiCallResponse = HttpSignedDataApiCallSuccessResponse | ApiCallErrorResponse;
+
+export type ApiCallResponse = RegularApiCallResponse | HttpGatewayApiCallResponse | HttpSignedDataApiCallResponse;
+export interface RegularApiCallSuccessResponse {
+  success: true;
+  data: { encodedValue: string; signature: string };
+}
 
-export interface ApiCallSuccessResponse {
+export interface HttpGatewayApiCallSuccessResponse {
   success: true;
-  value: string;
-  signature: string;
+  data: { values: unknown[]; rawValue: unknown; encodedValue: string };
```

Comment:
Because you can have multiple reserved parameters (https://docs.api3.org/ois/v1.0.0/reserved-parameters.html#encoding-multiple-values) and the `values` will be an array of each of these extracted values. The `rawValue` is (a single) full API response and `encodedValue` is an a hex string.

Reply:
The `rawValue` is a fair point but `encodedValue` is IMO confusing. I understand the result is one string but it can be potentially multiple values encoded in that string.

Reply:
Depends on perspective :D The chain expects just a single encoded value and does not care if it's just one or multiple values encoded in tuple.